searx/poolrequests.py

```diff
@@ -182,3 +191,60 @@ def patch(url, data=None, **kwargs):
 
 def delete(url, **kwargs):
     return request('delete', url, **kwargs)
+
+
+def dict_request(method, query, **kwargs):
```

Comment:
`poolrequests.py` could be renamed `https.py` or `http_client.py`, because it focuses on HTTP requests.

It would be better have a different file for the dict protocol.

Reply:
Some engines imports modules from poolrequests, so this requires a lot of changes
```
$ rg poolrequests
searx/autocomplete.py
28:from searx.poolrequests import get as http_get
searx/poolrequests.py
13:logger = logger.getChild('poolrequests')
searx/search.py
26:import searx.poolrequests as requests_lib
searx/engines/soundcloud.py
19:from searx.poolrequests import get as http_get
searx/engines/wolframalpha_noapi.py
15:from searx.poolrequests import get as http_get
searx/engines/wikidata.py
23:from searx.poolrequests import post, get
searx/engines/__init__.py
28:from searx.poolrequests import get
searx/engines/pubmed.py
18:from searx.poolrequests import get
searx/engines/gigablast.py
19:from searx.poolrequests import get
searx/engines/yggtorrent.py
16:from searx.poolrequests import get as http_get
searx/engines/duckduckgo_images.py
20:from searx.poolrequests import get
```

Separate `dict_client.py` shouldn't have any drawback.